core/src/main/java/io/onetable/client/TableFormatClientFactory.java

```diff
@@ -43,15 +38,21 @@ public static TableFormatClientFactory getInstance() {
 
   public TargetClient createForFormat(
       String tableFormat, PerTableConfig perTableConfig, Configuration configuration) {
-    switch (tableFormat) {
-      case ICEBERG:
-        return new IcebergClient(perTableConfig, configuration);
-      case DELTA:
-        return new DeltaClient(perTableConfig, configuration);
-      case HUDI:
-        return new HudiTargetClient(perTableConfig, configuration);
-      default:
-        throw new NotSupportedException("Target format is not yet supported: " + tableFormat);
+    TargetClient targetClient = createTargetClientForName(tableFormat);
+    if (targetClient == null) {
```

Comment:
Is it possible for `targetClient` to have a null value here?

Reply:
Yes. If there are no implementations that match the provided format name from those loaded by ServiceLoader.

Reply:
Would line 56 throw an exception if no match is found?

Reply:
Yep - you're right. I was thinking about the code before you asked me to not return null. :)

Reply:
done.